src/common/util.cc

```diff
@@ -268,7 +268,7 @@ void collect_sys_info(map<string, string> *m, CephContext *cct)
   // processor
   f = fopen(PROCPREFIX "/proc/cpuinfo", "r");
   if (f) {
-    char buf[1024];
+    char buf[1048576];
```

Comment:
i don't understand why this addresses https://tracker.ceph.com/issues/43306 . as fgets(3) does not segfault if the file being read is larger than the buffer. but i posted #32630 to avoid using fgets(3) anyway.

Reply:
maybe because the resulting buffer doesn't get null-terminated?

Reply:
no, per fgets(3)

>      fgets() reads in at most  one less than size characters from
>      stream  and stores  them into  the buffer  pointed to  by s.
>      Reading stops  after an EOF or  a newline.  If a  newline is
>      read, it is stored into the buffer.  A terminating null byte
>      ('\0') is stored after the last character in the buffer.

so the resulting buffer is always null-terminated.